be/src/exec/orc_scanner.cpp

```diff
@@ -0,0 +1,328 @@
+    } catch (orc::ParseError &e) {
+        std::stringstream str_error;
+        str_error << "ParseError : " << e.what();
+        LOG(WARNING) << str_error.str();
+        return Status::InternalError(str_error.str());
+    } catch (orc::InvalidArgument &e) {
+        std::stringstream str_error;
+        str_error << "ParseError : " << e.what();
+        LOG(WARNING) << str_error.str();
+        return Status::InternalError(str_error.str());
+    }
+}
+
+Status ORCScanner::open_next_reader() {
+    while (true) {
+        if (_next_range >= _ranges.size()) {
+            _scanner_eof = true;
+            return Status::OK();
+        }
+        const TBrokerRangeDesc &range = _ranges[_next_range++];
+        std::unique_ptr<FileReader> file_reader;
+        switch (range.file_type) {
+            case TFileType::FILE_LOCAL: {
+                file_reader.reset(new LocalFileReader(range.path, range.start_offset));
+                break;
+            }
+            case TFileType::FILE_BROKER: {
+                int64_t file_size = 0;
+                // for compatibility
+                if (range.__isset.file_size) { file_size = range.file_size; }
+                file_reader.reset(new BrokerReader(_state->exec_env(), _broker_addresses, _params.properties,
+                                                   range.path, range.start_offset, file_size));
+                break;
+            }
+            default: {
+                std::stringstream ss;
+                ss << "Unknown file type, type=" << range.file_type;
+                return Status::InternalError(ss.str());
+            }
+        }
+        RETURN_IF_ERROR(file_reader->open());
+        if (file_reader->size() == 0) {
+            file_reader->close();
+            continue;
+        }
+
+        std::unique_ptr<orc::InputStream> inStream = std::unique_ptr<orc::InputStream>(
+                new ORCFileStream(file_reader.release(), range.path));
+        _reader = orc::createReader(std::move(inStream), _options);
+
+        _total_groups = _reader->getNumberOfStripes();
+        _current_group = 0;
+        _rows_of_group = 0;
+        _current_line_of_group = 0;
+
+        _includes.clear();
+        _num_of_columns_from_file =
+                range.__isset.num_of_columns_from_file ? range.num_of_columns_from_file : _src_slot_descs.size();
+        for (int i = 0; i < _num_of_columns_from_file; i++) {
+            auto slot_desc = _src_slot_descs.at(i);
+            _includes.push_back(slot_desc->col_name());
+        }
+        _rowReaderOptions.include(_includes);
+
+        int orc_index = 0;
+        std::map<std::string, int> column_name_map_orc_index;
+        for (int i = 0; i < _reader->getType().getSubtypeCount(); ++i) {
+            if (std::find(_includes.begin(), _includes.end(), _reader->getType().getFieldName(i)) != _includes.end()) {
+                column_name_map_orc_index.emplace(_reader->getType().getFieldName(i), orc_index++);
+            }
```

Comment:
else should return bad status

Reply:
include columns must in reader field, otherwise createRowReader will throw exception